internal/app/networkd/pkg/networkd/misc.go

```diff
@@ -13,26 +13,47 @@ import (
 	"strings"
 	"text/template"
 
+	"github.com/jsimonetti/rtnetlink"
+	"golang.org/x/sys/unix"
+
 	talosnet "github.com/talos-systems/talos/pkg/net"
 )
 
-// filterInterfaceByName filters network links by name so we only mange links
+// filterInterfaces filters network links by name so we only mange links
 // we need to.
-func filterInterfaceByName(links []net.Interface) (filteredLinks []net.Interface) {
-	for _, link := range links {
+func filterInterfaces(interfaces []net.Interface) (filtered []net.Interface, err error) {
+	var conn *rtnetlink.Conn
+	conn, err = rtnetlink.Dial(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, iface := range interfaces {
+		link, err := conn.Link.Get(uint32(iface.Index))
+		if err != nil {
+			return nil, err
```

Comment:
should we skip those links which give back an error? as otherwise this fails the whole flow?